sdk/resource/host_id_bsd.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build dragonfly || freebsd || netbsd || openbsd || solaris
+// +build dragonfly freebsd netbsd openbsd solaris
+
+package resource // import "go.opentelemetry.io/otel/sdk/resource"
+
+import (
+	"errors"
+	"strings"
+)
+
+// hostIDReaderBSD implements hostIDReader
+type hostIDReaderBSD struct {
+	execCommand commandExecutor
+	readFile    fileReader
+}
+
+// read attempts to read the machine-id from /etc/hostid. If not found it will
+// execute `kenv -q smbios.system.uuid`. If neither location yields an id an
+// error will be returned.
+func (r *hostIDReaderBSD) read() (string, error) {
```

Comment:
I can move the readers out of the platform specific files and only leave the initialization of `var platformHostIDReader hostIDReader = ...` in each. This would be possible for everything but windows anyways.

Reply:
Let me know if moving the platform specific code is what you had in mind, or if you had more a refactor to structure of the readers. As it stands, the readers get their data by running shell commands, reading files, a combination of both, in the case of windows neither is true as it reads from the registry. The darwin reader has complex parsing, the other readers only need to trim whitespace from the files or commands executed.

Reply:
I went ahead and moved `hostIDReaderBSD`, `hostIDReaderDarwin`, and `hostIDReaderLinux` into `host_id.go` with no build flags. I moved their respective tests into `host_id_reader_test.go` and they now run on all platforms. I left `hostIDReaderWindows` and `hostIDReaderUnsupported` as is. I think this addresses at least some of your concerns. Let me know if you had more changes in mind.